node/src/main/scala/coop/rchain/node/NodeRuntime.scala

```diff
@@ -241,13 +241,18 @@ class NodeRuntime private[node] (
         } yield ()
       else Task.unit
 
-    val loop: Task[Unit] =
+    val nodeDiscoveryLoop: Task[Unit] =
       for {
-        _ <- dynamicIpCheck
         _ <- NodeDiscovery[Task].discover
+        _ <- time.sleep(20.seconds)
+      } yield ()
+
+    val clearConnectionsLoop: Task[Unit] =
+      for {
+        _ <- dynamicIpCheck
         _ <- Connect.clearConnections[Task]
         _ <- Connect.findAndConnect[Task](Connect.connect[Task])
```

Comment:
Please move `Connect.findAndConnect()` to the `nodeDiscoveryLoop` after `NodeDiscovery[Task].discover`